Frameworks/Blockcerts/CertificateV3.swift

```diff
@@ -0,0 +1,264 @@
+        }
+        
+        var certificateTitle = ""
+        var certificateDescription = ""
+        
+        if let credentialSubject = json["credentialSubject"] as? [String : AnyObject],
+            let claim = credentialSubject["claim"] as? [String: AnyObject] {
+            certificateTitle = claim["name"] as? String ?? ""
+            certificateDescription = claim["description"] as? String ?? ""
+        }
+        
+        self.title = certificateTitle
+        self.description = certificateDescription
+        
+        id = certificateIdString
+        shareUrl = URL(string: id)
+        language = ""
+        
+        self.subtitle = ""
+        self.image = imageData(from: "")
+        
+        // Use helper methods to parse Issuer, CredentialSubject and Display.
+        guard let issuer = MethodsForV3.parse(issuerJSON: json["issuer"]),
+              let recipient = MethodsForV3.parse(recipientJSON: json),
+              let display = MethodsForV3.parse(displayJSON: json["display"]),
+              let metadataJSON = MethodsForV3.parse(metadataJSON: json),
+              let assertion = MethodsForV3.parse(assertionJSON: assertionVal as AnyObject?),
+              let verifyData = MethodsForV3.parse(verifyJSON: json) else {
+            throw CertificateParserError.genericError
+        }
+        
+//        guard let receiptData = MethodsForV3.parse(receiptJSON: json["signature"]) else {
+//            throw CertificateParserError.genericError
+//        }
+    
+        self.issuer = issuer
+        self.recipient = recipient
+        self.assertion = assertion
+        self.verifyData = verifyData
+        self.receipt = nil // receiptData
+        self.metadata = metadataJSON
+        self.htmlDisplay = display
+        self.signature = nil
+        self.credentialSubject = json["credentialSubject"] as! [String : AnyObject]
+        self.issuanceDate = json["issuanceDate"] as! String
+        self.expirationDate = json["expirationDate"] as? String ?? ""
+    }
+}
+
+fileprivate enum MethodsForV3 {
+    static func parse(issuerJSON: AnyObject?) -> Issuer? {
+        // TODO support issuer as string or as object
+        var issuerId: String
+        
+        if issuerJSON is String {
+            issuerId = issuerJSON as! String
+            guard let issuerIdURL = URL(string: issuerId) else {
+                print("Issuer not defined")
+                return nil
+            }
+            return IssuerV2(id: issuerIdURL)
+        } else {
+            issuerId = issuerJSON?["id"] as! String
+        }
+        
+        guard let issuerData = issuerJSON as? [String : String],
+            let issuerURLString = issuerData["url"],
+            let issuerURL = URL(string: issuerURLString),
+            let logoURI = issuerData["image"],
+            let issuerEmail = issuerData["email"],
+            let issuerName = issuerData["name"],
+            let issuerDescription = issuerData["description"],
```

Comment:
This variable is not used. Please ignore if it is intensional?

Reply:
👍